executor/builder.go

```diff
@@ -1088,6 +1088,14 @@ func (b *executorBuilder) buildUnionScanFromReader(reader Executor, v *plannerco
 		us.columns = x.columns
 		us.table = x.table
 		us.virtualColumnIndex = buildVirtualColumnIndex(us.Schema(), us.columns)
+	case *IndexMergeReaderExecutor:
+		// IndexMergeReader doesn't care order.
```

Comment:
But, the CNF form may care order. We need to take it carefully, not to forget add it here(

Reply:
Sorry I don't understand why we care CNF. If the required prop needs to be [ordered](https://github.com/pingcap/tidb/blob/b1352ded030c33065e711440939b97767a518d76/planner/core/find_best_task.go#L961), we will not generate IndexMergeReader?

Reply:
For the DNF case, we cannot maintain the order since it's hard to maintain the order for `a = 1 or b = 1 order by a`.
But for CNF case `a > 1 and b > 1 order by a`. Is possible that we choose the a as a primary path to keep the order.

Reply:
Really sorry for the late reply. IMHO in the CNF case, a sort operator should be added instead of choosing `a` as the primary key for sorting?

Because according to the [current implementation](https://github.com/pingcap/tidb/blob/b1352ded030c33065e711440939b97767a518d76/planner/core/find_best_task.go#L961), if the `prop` has sorting requirements, `convertToIndexMergeScan` will directly return invalidTask. Then the optimizer should clear the prop then choosing the index merge again, and finally a sort operator will be added? Is that the process?